src/main/java/CustomExtendedThread.java

```diff
@@ -0,0 +1,16 @@
+import org.apache.log4j.Logger;
+
+public class CustomExtendedThread extends Thread {
+    private static final Logger logger = Logger.getLogger(CustomExtendedThread.class);
+    private final Counter counter;
+
+    public CustomExtendedThread(Counter counter) {
+        this.counter = counter;
+    }
+
+    public void run() {
+        for (int i = 0; i <= 100; i++) {
```

Comment:
```suggestion
        while (counter.getCount() < 100) {
```
you shouldn't perform this operation 100 times, just increment it till certain value

Reply:
refactored to use with `while` loop